### DISCUSSION TITLE
[NonNASA] decoding raw data  ? where and how ?

### DISCUSSION THREAD

**Discussion by tfoutfou** [omerfaruk-aran/esphome_samsung_hvac_bus#171] [Questions]
Hi ,

i have looked at `protocol_non_nasa.h`  and `protocol_non_nasa.cpp`      it seem all the decoding more or less happen here

my question are : 
-where did you find that information ?
-can you share the source or how you get it ?
-is it your work by trial and error and packet sniffing ? (very impressive if this is it)
-decompile Snet Pro V1 ?

ultimately my goal is to make my own code because i dont really need all the cumbersome (and very complicated for me to fully understand) code of ESPhome and to run it on an ESP32 that rely on wifi. 
I plan to run a smaller version for my need that run on an Arduino Mega2560 wich i believe is more than enough to process 10 packet/second max @ 9600bps

Thank you very much for your time

**Comment by lanwin** [comment]
Most of the code is a reverse engineered from Samsungs SNET Software. SNET is written in dotnet and dotnet is not binary code. So its easy to find Discompilers and read what it does.

**Reply by tfoutfou** [reply]
is it your work directly , did you decompile it to write code for this repo ?

can you share your findings and some documentation or do i have to look at it myself ? ( don t know at all how to decompile and deal with .Net)  will do some research but it may be well above my head

**Reply by lanwin** [reply]
No I will not do this, simply cause its to dangerous for legal reasons. 

And sorry, but I consider the code my documentation. For everything else I did not have the time.
